### PR TITLE
add collaboration accounts tutorial

### PR DESCRIPTION
**Pull request by minrk**
I tried to write up the collaboration accounts example as a tutorial. I think there's still some polish to do and add some screenshots, but this is what I have before I need to quit for the day.

cc @fperez @ryanlovett

### CODE REVIEW

File: docs/source/tutorial/collaboration-users.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
+# Real-time collaboration without impersonation
+
+:::{note}
+It is recommended to use at least JupyterLab 3.6 with JupyterHub >= 3.1.1 for this.
+:::
+
+JupyterLab has support for real-time collaboration, where multiple users are working with the same Jupyter server and see each others edits.
```
Comment (fperez) [inline docs/source/tutorial/collaboration-users.md:7]: s/others/other's/ ? (Not sure, not native speaker).

File: docs/source/tutorial/collaboration-users.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+Beyond other collaborative-editing environments, Jupyter includes _execution_.
+So granting someone access to your server also means granting them access to **run code as you**.
+That's a pretty big difference, and may not be acceptable to your users (or sysadmins!).
+
+One strategy for this is to have the concept of "collaboration accounts", function as users themselves and represent the _collaboration_ instead of any individual human.
```
Comment (fperez) [inline docs/source/tutorial/collaboration-users.md:12]: s/, function/, which function/

File: docs/source/tutorial/collaboration-users.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+```
+
+where we create the `collaborative` group which will contain only the collaboration accounts themselves.
+
+### Creating collaboration accounts
+
+Next, we are going to iterate through our collaborations, create users, and assign permissions.
+We are going to:
+
+1. create a user for each collaboration
```
Comment (fperez) [inline docs/source/tutorial/collaboration-users.md:85]: s/user/jupyterhub user/

**Review by fperez — APPROVED** [review]
Very minor suggestions, this looks excellent. Huge thanks @minrk for taking the time to write it up!

**Comment by minrk** [comment]
Thanks for reading! I applied the suggestions and added some screenshots and a bit more detail.

**Review by fperez — APPROVED** [review]
Thx @minrk! I vote we merge this for now and we can iterate later on refining it after folks like @ryanlovett start having more experience using this at Berkeley...

**Comment by ryanlovett** [comment]
@minrk A few comments... not sure whether to create an issue or not:

1. The roles work great! Non-admin users who are members of the collab groups can spawn the associated shared account. They can also visit /hub/admin and see the shared accounts.

2. In my latest attempts, with `c.Authenticator.manage_groups=False`, the hub raised an exception when attempting to `c.JupyterHub.load_groups[foo] = bar`. The traceback below seems to be the error for when a value is not configurable. I worked around it by manually creating the groups and adding members under /hub/admin.

```
Add group 'shared-1' with members: ['user1', 'user2', 'user3']
[E 2023-02-24 18:40:39.130 JupyterHub app:3334]
    Traceback (most recent call last):
      File "/usr/local/lib/python3.11/site-packages/jupyterhub/app.py", line 3331, in launch_instance_async
        await self.initialize(argv)
      File "/usr/local/lib/python3.11/site-packages/jupyterhub/app.py", line 2808, in initialize
        self.load_config_file(self.config_file)
      File "/usr/local/lib/python3.11/site-packages/traitlets/config/application.py", line 113, in inner
        return method(app, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/local/lib/python3.11/site-packages/traitlets/config/application.py", line 899, in load_config_file
        for (config, fname) in self._load_config_files(
      File "/usr/local/lib/python3.11/site-packages/traitlets/config/application.py", line 858, in _load_config_files
        config = loader.load_config()
                 ^^^^^^^^^^^^^^^^^^^^
      File "/usr/local/lib/python3.11/site-packages/traitlets/config/loader.py", line 626, in load_config
        self._read_file_as_dict()
      File "/usr/local/lib/python3.11/site-packages/traitlets/config/loader.py", line 659, in _read_file_as_dict
        exec(compile(f.read(), conf_filename, "exec"), namespace, namespace)  # noqa
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/local/etc/jupyterhub/jupyterhub_config.py", line 489, in <module>
        exec(config_py)
      File "<string>", line 24, in <module>
    TypeError: 'LazyConfigValue' object does not support item assignment
```

3. Our shared usernames are of the form `collab-shared-N` and the groups are `shared-N`. I tried to spawn a shared server by going to /hub/spawn/collab-shared-N, but it was 404. I then manually created the collab-shared-N user and then was able to spawn.


As context, our z2jh-based hub has a jupyterhub.hub.extraConfig stanza:
```
    extraConfig:
        20-define-collab-groups: |
            collab_groups = {
                "shared-1": [
                    "user1",
                    "user2",
                    "user3",
                ],
                "shared-2": [
                    "user1",
                    "user2",
                    "user4",
                ],
            }
            for collab_group, members in collab_groups.items():
                print(f"Add group '{collab_group}' with members: {members}")
                c.JupyterHub.load_groups[collab_group] = members

                # Add user associated with collab group
                collab_user = f"collab-{collab_group}"

                # Anyone who in a collab_group can admin and access the
                # user server, and can list users, of that group.
                print(f"Add role for '{collab_user}'")
                c.JupyterHub.load_roles.append(
                    {
                        "name": f"collab-access-{collab_group}",
                        "scopes": [
                            "admin-ui",
                            f"list:users!user={collab_user}",
                            f"admin:servers!user={collab_user}",
                            f"access:servers!user={collab_user}",
                        ],
                        "groups": [collab_group],
                    }
                )
```

**Pull request merged**